### PR TITLE
Add Encoding SSZ Package

### PR DESCRIPTION
**Pull request by rauljordan**
Similar to #9582, this PR creates encoding/ssz

### CODE REVIEW

File: shared/params/loader.go [hunk 1]
```diff
@@ -50,7 +50,8 @@ func LoadChainConfigFile(chainConfigFileName string) {
 	OverrideBeaconConfig(conf)
 }
 
-func replaceHexStringWithYAMLFormat(line string) []string {
+// ReplaceHexStringWithYAMLFormat will replace hex strings that the yaml parser will understand.
+func ReplaceHexStringWithYAMLFormat(line string) []string {
```
Comment (rkapka) [inline shared/params/loader.go:54]: Looks like this is exported only to satisfy tests which have been moved to `params_test` package. Is this OK?

File: encoding/ssz/htrutils.go [hunk 2]
```diff
@@ -56,22 +55,22 @@ func CheckpointRoot(hasher HashFn, checkpoint *ethpb.Checkpoint) ([32]byte, erro
...
 
-// HistoricalRootsRoot computes the HashTreeRoot Merkleization of
-// a list of [32]byte historical block roots according to the Ethereum
-// Simple Serialize specification.
-func HistoricalRootsRoot(historicalRoots [][]byte) ([32]byte, error) {
-	result, err := BitwiseMerkleize(hash.CustomSHA256Hasher(), historicalRoots, uint64(len(historicalRoots)), params.BeaconConfig().HistoricalRootsLimit)
+// ByteArrayRootWithLimit computes the HashTreeRoot Merkleization of
+// a list of [32]byte roots according to the Ethereum Simple Serialize
+// specification.
+func ByteArrayRootWithLimit(roots [][]byte, limit uint64) ([32]byte, error) {
```
Comment (rkapka) [inline encoding/ssz/htrutils.go:61]: Looks OK, but it is a little bit outside of the scope of a refactor PR.

File: shared/params/loader.go (hunk 1 above)
Comment (prestonvanloon) [reply shared/params/loader.go:54]: Yes

File: encoding/ssz/htrutils.go (hunk 2 above)
Comment (prestonvanloon) [reply encoding/ssz/htrutils.go:61]: The reason was that this package had a dependency on the params package. 
The change was to reduce the likelihood of a circular dependency in tests and make this library more generic.

**Pull request merged**